doc/rfcs/003-quickly-collaborate-with-coworkers-and-friends.md

```diff
@@ -0,0 +1,146 @@
+
+- Changing your online/offline status
+    - Setting your status to offline. (In the meantime, you can sign out of Teletype or disable Teletype in Atom's package settings when you want to use Atom while not accepting invitations from your past collaborators.)
+    - Setting your status to "away" or "busy"
+- UX enhancements to the list of past collaborators
+    - Changing the sort order for your list of past collaborators (e.g., sort by how recently you've collaborated, sort or group by online/offline status)
+    - Limiting the size of your list of past collaborators. (In the meantime, you can remove past collaborators to reduce the size of the list.)
+    - Filtering your list of past collaborators
+    - Marking certain collaborators as "favorites"
+- Offering to install Teletype if you click on a portal URL and don't yet have Teletype installed ([#220](https://github.com/atom/teletype/issues/220))
+- Selecting a past collaborator and asking to join their portal
+- Selecting multiple past collaborators and inviting them to your portal simultaneously. (In the meantime, you can select a collaborator and invite them, then select another collaborator and invite them, and so on.)
+
+## Drawbacks
+
+- Some people might feel uneasy knowing that past collaborators can see if they are "online" (i.e., if they have Atom open and they're signed into Teletype).
+- Once Teletype provides a list of past collaborators, people may want to be able to chat with those collaborators from within Teletype. While we may eventually want to support chat, any chat-related functionality must exist in service of Teletype's primary vision of "making it as easy to code together as it is to code alone." We'll need to be diligent to avoid scope creep.
+- Depending on the technical solution we choose for implementing presence, we may incur increased server-side resource consumption and/or we may take on additional operational complexity.
+
+## Rationale and alternatives
+
+##### Why is this approach the best in the space of possible approaches?
+
+With the approach described above, we believe Teletype can provide a streamlined set-up process while also preventing harassment. In order for Teletype to prompt you to join someone's portal, you must first have collaborated with that person in the past (by sending your portal URL to them via a third-party service or by joining their portal via a URL they sent you via a third-party service) *and* that person must still exist in your list of past collaborators (i.e., you haven't removed them from your list of past collaborators). In other words, people that you've never collaborated with cannot cause portal invitations to appear in your editor. And people that you've collaborated with in the past, but don't want to receive invitations from in the future, cannot cause portal invitations to appear in your editor either.
```

Comment:
For a while I was questioning: "Why is the list of past collaborators even needed, when there is no audio support or some sort of text based chat inside Atom?" Because if I invite somebody through the past collaborators list, after they join, I still need to go to Slack, etc. to chat or start an audio call. And if I have to do that, then I could just invite people by pasting the URL there. So it would make more sense to have a roadmap like this: 1. Add invite with URLs 2. Add audio/chat 3. Add past collaborators list.

But here some use cases where a past collaborators list would be handy even without audio/chat:

- When collaborators are in the same room. A host can just invite them directly.
- In recurring 1:1 meetings or standups where people join a video chat anyways. Then a host doesn't have to do a detour and be like "hey, here is today's portal" and can just invite people directly.
- Some people might still use traditional phones or things like FaceTime where you can't paste a link.
- You might start talking to people without intending to use Teletype. But then when you do, both, host and guest can go straight into Atom and the guest doesn't have to wait for the URL to be pasted into a chat.
- Not sure how often people do it, but you kinda can "chat" in an editor by just writing code comments to each other. I think when Teletype launched, and somebody made an ID public, people were "chatting" just to test Teletype.

Ok, so yeah.. maybe still enough good reasons to already have a past collaborators list now.

Reply:
Great points, @simurai. Thanks for this thoughtful comment. 

e35483b1fa69c3ef1ef56f0c437000d4dcda934b adds a blurb regarding this topic.

Reply:
Second the value of this without voice. There are times where I *still* don't join a portal when I should because I don't want to perform the copy-and-paste tea ceremony. Simply clicking a button will make this way more accessible.